public/app/features/alerting/state/AlertingQueryRunner.test.ts

```diff
@@ -0,0 +1,41 @@
+import { BackendSrv } from 'app/core/services/backend_srv';
+import { ColdObservable } from 'rxjs/internal/testing/ColdObservable';
+import { TestScheduler } from 'rxjs/testing';
+import { AlertingQueryResponse, AlertingQueryRunner } from './AlertingQueryRunner';
+
+const testScheduler = new TestScheduler((actual, expected) => {
+  expect(actual).toEqual(expected);
+});
+
+describe('AlertingQueryRunner', () => {
+  it('should successfully return panel data by refId', () => {
+    testScheduler.run((helpers) => {
```

Comment:
Are we OK with adding marble tests? If so, I will continue on this structure and add tests to cover the entire flow of the AlertingQueryRunner.

Reply:
I think we prefer using expect(obs).toEmitValues or expect(obs).toEmitValuesWith. But if you have really complex observables (I can't see the need here) I would use marble tests.

Reply:
Yeah, I basically want to test the flow in the `AlertingQueryRunner` that covers cancelations, loading updates, errors etc. I wanted to get early feedback before spending to much time on this.